docs/logging-guidelines.md

```diff
@@ -0,0 +1,167 @@
+
+**Audience:**
+- Users
+
+**Examples:**
+- Uncaught exceptions/unhandled errors
+- Resources exhausted (eg. out of memory, out of storage space)
+
+### ERROR
+**Purpose:** 
+- Indicate that app is unable to perform the requested **action** and stops trying.
+
+**Audience:**
+- Users
+- Integrator developers
+- Core system developers
+
+**Examples:**
+- REST API and CLI command processing errors
+- Payment-related (eg. blockchain interaction) errors
+
+### WARN
+**Purpose:**
+- Indicate that app is gracefully handling an erratic situation and is able to continue with the requested action.
+
+**Audience:**
+- Users
+- Node owners/admins
+- Integrator developers
+- Core system developers
+
+**Examples:**
+- Errors in actions which will be retried
+- Invalid parameters, attributes, addresses which will be ignored/superseded by eg. default values, etc. 
+
+### INFO
+**Purpose:**
+- Inform about successful initialization and shutdown of app module/feature.
+- Indicate that app is performing a requested action/command/request.
+
+**Audience:**
+- Users
+- Node owners/admins
+- Integrator developers
+- Core system developers
+
+**Examples:**
+- Startup event of a significant module - including fundamental parameters of execution, eg. 
+  - URLs/port numbers for services listening or depending on network connectivity,
+  - Working directories, data directories 
+- Shutdown event of a significant module 
+- Command requested from the module
+  - Indication of a CLI command sent to Yagna daemon
+  - Indication of REST API request received
+  
+### DEBUG
+**Purpose:**
+- Provide additional context of performed actions, additional steps or any info which may be useful for troubleshooting.
+
+**Audience:**
+- Integrator developers
+- Core system developers
+- Node owners/admins 
+
+**Examples:**
+- Low level processing steps, especially those dependent on files, resources, connectivity, with dependency info (URLs, addresses, parameter values)
+- 
+
+### TRACE
+**Purpose:**
+- Record low-level/technical details of performed actions, like net traffic content (API requests, responses, Golem Net messages, etc.)
+
+**Audience:**
+- Core system developers
+
+**Examples:**
+- Environment variable/config parameter value snapshots
+- API request/response body content
+- Golem net message routing info and content
+
+## Error handling & logging guidelines
+
+As low-level, 3rd party library errors are encountered during execution, their error messages are usually useless without context information. It is vital to wrap the low-level error messages with additional info to indicate details of performed activity that would aid in troubleshooting. Eg. instead of:
+
+```
+[2020-08-27T07:56:22Z ERROR yagna] File IO error: Path not found
+```
+
+log this:
+
+```
+[2020-08-27T07:56:22Z ERROR yagna] E00342 - WASM ExeUnit DEPLOY: Error downloading remote file to temp folder '/tmp/yagna_data': File IO error: Path not found
```

Comment:
From my experience of debugging yagna: the file name and line number where the error occurred is very important as it's ofter easier to just read the source code to find the problem. (I know you mentioned it in "Log entry content" section.)

Reply:
Agreed, albeit the error code indicated in this example serves this purpose: with a specific error code it should be fairly instantaneous to identify the location responsible for a particular issue (I'm conscious that providing file/line number may not be feasible for many dev platforms).